### ISSUE TITLE
Players don't need to know their pieces

### ISSUE THREAD

**Issue by Gursach44** [CSC207-2022F-UofT/course-project-chmess#50]
On a (possibly erroneous) review of the program structure, it seems like players do not need to know which pieces they own. This is because any information about a specific piece can be accessed as long as the piece is available, since the piece knows its player's colour already.

**Comment by cherche** [comment]
I agree with your assessment. Up until a recent merge, this was the case. It would be best to refactor to return to such a state so that the Player and Piece classes are not so tightly coupled. It might even be good to move capturedPieces into Board, rather than to keep them in each Player.
